scylla/src/transport/connection_pool.rs

```diff
@@ -603,7 +605,9 @@ impl PoolRefiller {
                 }
 
                 evt = self.ready_connections.select_next_some(), if !self.ready_connections.is_empty() => {
-                    self.handle_ready_connection(evt);
+                    if let Ok(evt) = evt {
+                        self.handle_ready_connection(evt);
+                    }
```

Comment:
You must call `self.handle_ready_connection` in case of failure. It updates the state of the pool as a reaction to the error.

`OpenedConnectionEvent` already holds a result, so you should put the translation error there (probably converted to a QueryError?).

Reply:
Do you find `QueryError` suitable for containing variants regarding _translating addresses on establishing connection to nodes_ ? For me, this is highly counterintuitive. I'm getting more and more convinced that it's high time we considered refactor of our errors.

Reply:
After all, there wasn't any query attempted yet, and the user gets a `QueryError`.

Reply:
> Do you find `QueryError` suitable for containing variants regarding _translating addresses on establishing connection to nodes_ ? For me, this is highly counterintuitive. I'm getting more and more convinced that it's high time we considered refactor of our errors.

I agree that our error types need redesign, but that is not something that we can fix in this PR.

In this case though it wouldn't hurt to introduce an OpenConnectionError or sth. The error is converted to a string [here](https://github.com/scylladb/scylla-rust-driver/blob/4964b4471c3d14bfea75e532b2e7c24cbd48cd65/scylla/src/transport/connection_pool.rs#L348-L356) (again, not ideal), but the new error won't appear in the API and will slightly improve the clarity of the internal code.

> After all, there wasn't any query attempted yet, and the user gets a `QueryError`.

To be fair, using QueryError to represent an error related to a failed connection is not perfect, either.